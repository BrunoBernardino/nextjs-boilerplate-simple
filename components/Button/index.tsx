import React, { forwardRef } from 'react';
import styled from 'styled-components';

import { css } from 'lib/constants';

type ButtonProps = {
  onClick: any;
  children: React.ReactNode;
  className?: string;
  type?: 'button' | 'submit' | 'reset';
};

const StyledButton = styled.button`
  margin: 1em auto;
  padding: 1em 1.5em;
  font-size: 1.2em;
  border: none;
  border-radius: 3px;
  color: ${css.white};
  background: ${css.purple};
  cursor: pointer;

  &:hover,
  &:active,
  &:focus {
    background: ${css.pink};
  }
`;

const Button = forwardRef((props: ButtonProps, ref: any) => {
  const { onClick, children, className, type, ...remainingProps } = props;

  return (
    <StyledButton
      ref={ref}
      onClick={onClick}
      className={className}
      type={type}
      {...remainingProps}
    >
      {children}
    </StyledButton>
  );
});

Button.defaultProps = {
  type: 'button',
};

export default Button;
