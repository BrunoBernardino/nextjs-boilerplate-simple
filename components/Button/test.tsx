import React from 'react';
import renderer from 'react-test-renderer';
import { shallow } from 'enzyme';
import expect from 'expect';
import enzymify from 'expect-enzyme';

import Button from './index';

expect.extend(enzymify());

const defaultProps = {
  onClick: () => {},
};

describe('Button', () => {
  it('renders the button without errors', () => {
    const wrapper = shallow(<Button {...defaultProps}>Hello</Button>);
    expect(wrapper.first().text()).toBe('Hello');
  });

  it('renders the button as expected', () => {
    const tree = renderer
      .create(<Button {...defaultProps}>Hello</Button>)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
