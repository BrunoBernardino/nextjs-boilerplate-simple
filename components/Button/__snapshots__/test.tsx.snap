// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button renders the button as expected 1`] = `
<button
  className="Button__StyledButton-h5w9x6-0 bMGHFb"
  onClick={[Function]}
  type="button"
>
  Hello
</button>
`;
